File: backend/schemas/map.py
from pydantic import BaseModel


class HeatPoint(BaseModel):
    lat: float
    lng: float
    weight: float


class MapMarker(BaseModel):
    lat: float
    lng: float
    predicted_class: str
    color: str
    location_label: str | None
    created_at: str
    stored_image_url: str | None = None
    status: str | None = None


class HeatmapResponse(BaseModel):
    points: list[HeatPoint]
    total: int
    deterioration_count: int
    critical_count: int


class MarkersResponse(BaseModel):
    markers: list[MapMarker]

File: frontend/src/api/map.ts
import client from "./client";

export interface HeatPoint {
  lat: number;
  lng: number;
  weight: number;
}

export interface MapMarker {
  lat: number;
  lng: number;
  predicted_class: string;
  color: string;
  location_label: string | null;
  created_at: string;
  stored_image_url: string | null;
  status: string | null;
  suciedad_clase: string | null;
  deterioro_clase: string | null;
}

export interface HeatmapResponse {
  points: HeatPoint[];
  total: number;
  deterioration_count: number;
  critical_count: number;
}

export interface MarkersResponse {
  markers: MapMarker[];
}

export const mapApi = {
  heatmap: () => client.get<HeatmapResponse>("/map/heatmap").then((r) => r.data),
  markers: () => client.get<MarkersResponse>("/map/markers").then((r) => r.data),
};
